crates/nu-command/src/platform/timeout.rs

```diff
@@ -0,0 +1,150 @@
+use nu_engine::{command_prelude::*, get_eval_block_with_early_return};
+use nu_protocol::{engine::Closure, format_duration};
+use std::{sync::mpsc, thread, time::Duration};
+
+#[derive(Clone)]
+pub struct Timeout;
+
+impl Command for Timeout {
+    fn name(&self) -> &str {
+        "timeout"
+    }
+
+    fn description(&self) -> &str {
+        "Run a closure with a time limit."
+    }
+
+    fn extra_description(&self) -> &str {
+        r#"If the closure does not complete within the specified duration, it will be terminated
+and an error will be returned. This is useful for preventing long-running operations
+from blocking indefinitely, such as in MCP servers or automation scripts."#
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("timeout")
+            .input_output_types(vec![(Type::Nothing, Type::Any)])
```

Comment:
Are you should you want and input type of nothing? This would prevent being able to be used in a pipeline like this:
```nu
ps | timeout 5sec { $in | polars into-df } 
```

I would change it to Type::Any